addons/account_lock_constrains/__openerp__.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+{
+    'name' : 'Irreversible Lock Date',
+    'version' : '1.0',
+    'category': 'Accounting',
+    'description': """
+    Make the lock date irreversible:
+
+    - The lock date for advisors must be higher of equals of the lock date for users.
+    - The lock date for advisors must be higher of the last day of previous month.
+    - The lock date for advisors must be strictly higher of the previous one.
```

Comment:
error messages are wrong and should be fixed as suggested by Laura:
You cannot define stricter conditions on advisors than on users. Please make sure that the lock date on advisor is not set after the lock date for users.
You cannot lock a period that is not finished yet. Please make sure that the lock date for advisors is not set after the last day of the previous month.
The new lock date for advisors must be set after the previous lock date.

Please adapt the docstring, the error messages and this module's description accordingly